modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultIRTreeToASMBytesPhase.java

```diff
@@ -0,0 +1,1655 @@
+        MethodWriter methodWriter = writeScope.getMethodWriter();
+        methodWriter.writeStatementOffset(irForEachSubArrayNode.getLocation());
+
+        Variable variable = writeScope.defineVariable(irForEachSubArrayNode.getVariableType(), irForEachSubArrayNode.getVariableName());
+        Variable array = writeScope.defineInternalVariable(irForEachSubArrayNode.getArrayType(), irForEachSubArrayNode.getArrayName());
+        Variable index = writeScope.defineInternalVariable(irForEachSubArrayNode.getIndexType(), irForEachSubArrayNode.getIndexName());
+
+        visit(irForEachSubArrayNode.getConditionNode(), writeScope);
+        methodWriter.visitVarInsn(array.getAsmType().getOpcode(Opcodes.ISTORE), array.getSlot());
+        methodWriter.push(-1);
+        methodWriter.visitVarInsn(index.getAsmType().getOpcode(Opcodes.ISTORE), index.getSlot());
+
+        Label begin = new Label();
+        Label end = new Label();
+
+        methodWriter.mark(begin);
+
+        methodWriter.visitIincInsn(index.getSlot(), 1);
+        methodWriter.visitVarInsn(index.getAsmType().getOpcode(Opcodes.ILOAD), index.getSlot());
+        methodWriter.visitVarInsn(array.getAsmType().getOpcode(Opcodes.ILOAD), array.getSlot());
+        methodWriter.arrayLength();
+        methodWriter.ifICmp(MethodWriter.GE, end);
+
+        methodWriter.visitVarInsn(array.getAsmType().getOpcode(Opcodes.ILOAD), array.getSlot());
+        methodWriter.visitVarInsn(index.getAsmType().getOpcode(Opcodes.ILOAD), index.getSlot());
+        methodWriter.arrayLoad(MethodWriter.getType(irForEachSubArrayNode.getIndexedType()));
+        methodWriter.writeCast(irForEachSubArrayNode.getCast());
+        methodWriter.visitVarInsn(variable.getAsmType().getOpcode(Opcodes.ISTORE), variable.getSlot());
+
+        Variable loop = writeScope.getInternalVariable("loop");
+
+        if (loop != null) {
+            methodWriter.writeLoopCounter(loop.getSlot(), irForEachSubArrayNode.getLocation());
+        }
+
+        visit(irForEachSubArrayNode.getBlockNode(), writeScope.newLoopScope(begin, end));
+
+        methodWriter.goTo(begin);
+        methodWriter.mark(end);
+    }
+
+    @Override
+    public void visitForEachSubIterableLoop(ForEachSubIterableNode irForEachSubIterableNode, WriteScope writeScope) {
+        MethodWriter methodWriter = writeScope.getMethodWriter();
+        methodWriter.writeStatementOffset(irForEachSubIterableNode.getLocation());
+
+        Variable variable = writeScope.defineVariable(
+                irForEachSubIterableNode.getVariableType(), irForEachSubIterableNode.getVariableName());
+        Variable iterator = writeScope.defineInternalVariable(
+                irForEachSubIterableNode.getIteratorType(), irForEachSubIterableNode.getIteratorName());
+
+        visit(irForEachSubIterableNode.getConditionNode(), writeScope);
+
+        if (irForEachSubIterableNode.getMethod() == null) {
+            org.objectweb.asm.Type methodType = org.objectweb.asm.Type
```

Comment:
Break this out, there's a lot of text in this statement.

Reply:
Cleaned this up by importing the Type class from org.objectweb.asm since there's no longer a conflict.